drivers/docker/driver.go

```diff
@@ -1461,7 +1461,7 @@ func (d *Driver) SignalTask(taskID string, signal string) error {
 		return fmt.Errorf("failed to parse signal: %v", err)
 	}
 
-	return h.Signal(sig)
+	return h.Signal(context.Background(), sig)
```

Comment:
This codepath is triggered by user requests, so I'm intentionally leaving the existing behavior here so we can review all the Docker API timeouts later. I only want to address the task kill issue.

Reply:
SGTM. Mind leaving that in a code comment?

Reply:
Done, including a link to https://github.com/hashicorp/nomad/issues/9503